### ISSUE TITLE
Error 137 while installing requirements

### ISSUE THREAD

**Issue by kostrykin** [galaxyproject/training-material#4313]
The [GTN tutorial on viewing the site locally](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/running-jekyll/tutorial.html) basically says that one should run
1. `make create-env`
2. `make install`

after installing Conda.

I have a fresh local clone of the repository. When running `make create-env` (step 1), an out of memory error occurs (error 137):

> (base) user@hostname:~/Documents/galaxy-training-material$ make create-env
> find: ‘_site/training-material’: No such file or directory
> find: ‘_site/training-material/*/*/slides/*’: No such file or directory
> find: ‘_site/training-material’: No such file or directory
> if ~/.anaconda3/bin/conda env list | grep '^galaxy_training_material'; then \
>     ~/.anaconda3/bin/conda env update -f environment.yml; \
> else \
>     ~/.anaconda3/bin/conda env create -f environment.yml; \
> fi
> Collecting package metadata (repodata.json): done
> Solving environment: \ bash: line 4: 3332886 Killed                  ~/.anaconda3/bin/conda env create -f environment.yml
> make: *** [Makefile:43: create-env] Error 137

The same happens when I try running `make install` (step 2) after creating the environment manually using `mamba env create -f environment.yml`:

> (galaxy_training_material) user@hostname:~/Documents/galaxy-training-material$ make install
> find: ‘_site/training-material’: No such file or directory
> find: ‘_site/training-material/*/*/slides/*’: No such file or directory
> find: ‘_site/training-material’: No such file or directory
> if ~/.anaconda3/envs/galaxy_training_material/bin/conda env list | grep '^galaxy_training_material'; then \
>     ~/.anaconda3/envs/galaxy_training_material/bin/conda env update -f environment.yml; \
> else \
>     ~/.anaconda3/envs/galaxy_training_material/bin/conda env create -f environment.yml; \
> fi
> Retrieving notices: ...working... done
> Collecting package metadata (repodata.json): done
> Solving environment: \ bash: line 4: 3341561 Killed                  ~/.anaconda3/envs/galaxy_training_material/bin/conda env create -f environment.yml

So how to install the requirements?

**Comment by kostrykin** [comment]
So in case someone else stumbles into this issue. Running `make install --dry-run` gave me the necessary hints to install the requirements manually:

1. `gem update --no-document --system`
2. `ICONV_LIBS="-L~/.anaconda3/envs/galaxy_training_material/lib/ -liconv" gem install --no-document addressable:'2.5.2' jekyll jekyll-feed jekyll-redirect-from csl-styles awesome_bot html-proofer pkg-config kwalify bibtex-ruby citeproc-ruby fastimage`
3. `pushd ~/.anaconda3/envs/galaxy_training_material/share/rubygems/bin`
4. `ln -sf ../../../bin/ruby ruby`

However, for some reason, running `make serve-quick` afterwards fails with the error
> EnvironmentNameNotFound: Could not find conda environment: galaxy_training_material
> You can list all discoverable environments with `conda info --envs`.

although I can see from `conda info --envs` that the environment is there.

So instead of using `make serve-quick`, one has to use:
```
mv Gemfile Gemfile.backup || true && \
mv Gemfile.lock Gemfile.lock.backup || true && \
jekyll serve --strict_front_matter -d _site/training-material --incremental --config _config.yml,_config-dev.yml -P 4000 -H 0.0.0.0 ""
```

However, I guess this is rather a workaround than a solution, so I'll leave this issue as open.

**Comment by hexylena** [comment]
> an out of memory error occurs (error 137):

how much memory is on your system? It make sense that conda requires too much, but there is essentially no possible workaround there other than "buy more RAM".

As for using a non-conda installation, yes, this works well. A number of us use RVM to manage multiple Ruby installations, and build it that way since conda can indeed be quite 'heavy'

**Comment by kostrykin** [comment]
> how much memory is on your system? It make sense that conda requires too much, but there is essentially no possible workaround there other than "buy more RAM".

The system has 32 GiB of RAM. It‘s actually somewhat hard to believe that this isn’t supposed to be more than enough?

**Comment by hexylena** [comment]
Oh, no, that is much more than enough. It should easily build with ~4GB. In that case I'm really not sure why conda is having an out of memory issue there, generally that should only affect small machines. https://github.com/conda/conda/issues/7614

**Comment by kostrykin** [comment]
fixed by #4511
